userguide/tutorials/plugin_development.adoc

```diff
@@ -454,29 +454,40 @@ Sometimes, you may see the `org.apache.shiro.authz.UnauthenticatedException: Thi
 
 === Maven Build Error
 
-Sometimes, when you run `mvn clean install` on the plugin code, you may see the following Maven error:
+Sometimes, when you build the code using `mvn clean install` you may see some build errors as follows:
 
+* Enforcer error:
++
 [source,bash]
 ----
 Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (default) on project killbill-plugin-momo: Some Enforcer rules have failed.
 ----
-
-This generally occurs when your `pom.xml` contains dependencies that are not recommended. The maven build has lots of checks in place to make sure the right dependencies are pulled in, there are no duplicate dependencies, there are no obvious bugs, etc.
-If you'd like to ignore all these checks and still proceed with the build, you can run the following command:
-
++
+This generally occurs when your `pom.xml` contains dependencies that are not recommended. In such cases, we recommend that you fix the POM file and rerun the build or run the build with the `-Dcheck.fail-enforcer=false` option.
++
+* Spotbugs error: 
+Kill Bill automatically performs `spotbugs` checks when running `mvn verify`. So, this may result in some https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html[spotbugs] errors.
++
+In such cases we recommend you to fix the spotbugs errors and rerun the build or run the build with the `-Dcheck.fail-spotbugs=false` option.
++
+* Other errors:
+The build process has several checks in place to make sure the right dependencies are pulled in, there are no duplicate dependencies, there are no obvious bugs, etc. If you'd like to ignore all these checks and still proceed with the build, you can run the following command:
++
 [source,bash]
 ----
 mvn clean install -Dcheck.fail-enforcer=false -Dcheck.fail-dependency=false -Dcheck.fail-dependency-scope=false -Dcheck.fail-dependency-versions=false -Dcheck.fail-duplicate-finder=false -Dcheck.fail-enforcer=false -Dcheck.fail-spotbugs=false -Dcheck.ignore-rat=true
 ----
 
-However, this is *not recommended*, we recommend that you fix the POM file and run the build with all the checks in place.
-
 === java.lang.NoClassDefFoundError or java.lang.ClassNotFoundException
 
 Sometimes, when you develop a custom plugin, a `java.lang.NoClassDefFoundError` or a `java.lang.ClassNotFoundException` exception may occur on starting Kill Bill. This is most probably because the class in question is not present on the classpath. 
 
 For a plugin to work, any classes used by the plugin must be present on the classpath. So, the class needs to be present in the *plugin jar* itself or it must be *imported* from Kill Bill. Refer to the <<importing_additional_packages_in_plugins, "Importing Additional Packages in Plugins">> section above for further details.
 
+=== Payment plugin xxx is not registered
```

Comment:
```suggestion
=== Error "Payment plugin xxx is not registered"
```
I'm suggesting this, but maybe there's a better way to make it clear it's an error message.

Reply:
I like `Error "Payment plugin xxx is not registered"` but I've used `"Payment plugin xxx is not registered" Error` to keep this consistent with the other headings in the FAQ section.

Reply:
Sound good